include/invert_quda.h

```diff
@@ -563,6 +563,21 @@ namespace quda {
     */
     void PrintSummary(const char *name, int k, double r2, double b2, double r2_tol, double hq_tol);
 
+    /**
+       @brief Helper function which returns the epsilon tolerance for a given precision
+    */
+    const double solverPrecisionEpsilonHelper(QudaPrecision prec) const;
+
+    /**
+       @brief Returns the epsilon tolerance of the solver precision
+    */
+    const double solverPrecisionEpsilon() const;
+
+    /**
+       @brief Returns the epsilon tolerance of the solver sloppy precision
+    */
+    const double solverPrecisionSloppyEpsilon() const;
+
```

Comment:
Some comments here:
* The `solver` prefixes are redundant since we are in the `Solver` class here, so I think these should be removed.
* These functions return by value, so the `const` is redundant.  In fact the additional `const` on the return type will lead to warnings with some compilers when additional warnings are enabled.
* Instead of having separate `solverPrecisionEpsilonHelper` and `solverPrecisionEpsilon` functions, how about fusing these into a single function `double precisionEpsilon(QudaPrecision prec = QUDA_INVALID_PRECISION)`, and if the default then return the epsilon tolerance of the solver precision else return the epsilon of the argument.

Reply:
https://github.com/lattice/quda/pull/1111/commits/c64a21fc736d8d6e413267d63d02601cb5d52002 and https://github.com/lattice/quda/pull/1111/commits/408e985252d476e40932774b3a0bd33d44e4fb66